nautobot/core/tables.py

```diff
@@ -145,13 +145,18 @@ def __init__(
         else:
             if user is not None and not isinstance(user, AnonymousUser):
                 columns = user.get_config(f"tables.{self.__class__.__name__}.columns")
+        self.sequence = []
         if columns:
             for name, column in self.base_columns.items():
                 if name in columns and name not in self.exclude:
                     self.columns.show(name)
                 else:
                     self.columns.hide(name)
             self.sequence = [c for c in columns if c in self.base_columns]
+        else:
+            for c in self.base_columns:
+                if c not in self.exclude:
+                    self.sequence.append(c)
```

Comment:
This block feels "off" to me as I look at it. `self.sequence` is the *order* columns are listed in, which should be independent of which columns are visible or hidden. The `if columns` case makes some sense to me, as it is using the user preferences to both 1) reorder the sequence and 2) show/hide columns as per the user preferences, but it seems wrong to me to recalculate `self.sequence` in the `else` case. 

I think @gsnider2195 analysis in #6661 is correct in stating that:

> I think core is accessing the table's sequence property incorrectly when columns.names() should be used.

as `Table.columns` *does* respect `self.exclude` whereas `Table.sequence` does not and should not. I think it's likely that both `configurable_columns()` and `visible_columns()` are implemented incorrectly as such.

Reply:
I think we should also try to understand how/why `dynamic_group_count` is getting incorrectly added to `self.sequence`, though.